### ISSUE TITLE
dotnet test --arch X64 doesn't fail on Mac arm64

### ISSUE THREAD

**Issue by MarcoRossignoli** [dotnet/sdk#21980]
### Describe the bug
When I ran `dotnet test --arch X64` on Mac M1 it should fail like for other verbs.
On my test it compiles a RID and create a folder like `osx-X64`

### To Reproduce

[TestProjectNetcore.zip](https://github.com/dotnet/sdk/files/7331968/TestProjectNetcore.zip)

Download above project and run

```
dotnet test --arch X64
```

### Further technical details
.NET SDK (reflecting any global.json):
 Version:   6.0.100-rc.2.21505.57
 Commit:    ab39070116

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  11.4
 OS Platform: Darwin
 RID:         osx.11.0-arm64
 Base Path:   /usr/local/share/dotnet/sdk/6.0.100-rc.2.21505.57/

Host (useful for support):
  Version: 6.0.0-rc.2.21480.5
  Commit:  6b11d64e7e

.NET SDKs installed:
  6.0.100-rc.2.21505.57 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.0-rc.2.21480.10 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.0-rc.2.21480.5 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

**Comment by MarcoRossignoli** [comment]
@rainersigwald can you help on this one?
We're flowing `property:RuntimeIdentifier` to msbuild.

**Comment by rainersigwald** [comment]
Sorry, I don't think I understand what you're asking about.

It looks like the SDK doesn't validate RID unless it's using an apphost or a couple of other conditions, from the snippet sfoslund linked:

https://github.com/dotnet/sdk/blob/877d0e65276fc8996c95f6f5aacabb59ae9cc6ae/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets#L123-L127

Are you asking the SDK to do more validation?

**Comment by MarcoRossignoli** [comment]
Yep I was wondering if it's possible extend validation in case of non apphost for the test sdk.

**Comment by JakkuSakura** [comment]
I use this as a workaround
```xml
<ItemGroup Condition="$([System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture) == 'Arm64'">
        <PackageReference Include="WebViewControl-Avalonia-ARM64" Version="3.120.5" />
</ItemGroup>
```
